### PR TITLE
fix: use URI.join to structure item URLs correctly

### PR DESCRIPTION
**Pull request by necojackarc**
Related to #1

### CODE REVIEW

File: lib/breadcrumbs_on_rails/json_ld/builder.rb [hunk 1]
```diff
@@ -19,11 +20,32 @@ def render_element(element, index)
...
+          URI.scheme_list[root_url_with_default_options.scheme.upcase].build(
+            host: root_url_with_default_options.host,
+            userinfo: root_url_with_default_options.userinfo,
+            port: root_url_with_default_options.port,
+          )
+
+        # The returned value of `compute_path(element)` may contain the default
+        # options as path or query parameters. This is why this value shouldn't
+        # be passed in `URI::Generic.build` as a `path` option directly.
+        URI.join(root_url, compute_path(element))
```
Comment (necojackarc) [inline lib/breadcrumbs_on_rails/json_ld/builder.rb]: I'm not sure we should return a URI object or a string object, but it works.
I believe `JSON.pretty_generate` internally calls `to_s` to each object.

File: lib/breadcrumbs_on_rails/json_ld/builder.rb [hunk 2]
```diff
@@ -19,11 +20,32 @@ def render_element(element, index)
           "@type" => "ListItem",
           "position" => index,
           "item" => {
-            "@id" => "#{@context.root_url.chop}#{compute_path(element)}",
+            "@id" => compute_url(element),
```
Comment (yasaichi) [inline lib/breadcrumbs_on_rails/json_ld/builder.rb]: [IMO]
I think the most simple solution is to use `URI.join` as follows:

```ruby
URI.join(@context.root_url, compute_path(element))
```

Comment (yasaichi) [inline lib/breadcrumbs_on_rails/json_ld/builder.rb]: [IMO]
I think the most simple solution is just to call `URI.join` as follows:

```ruby
URI.join(@context.root_url, compute_path(element))
```

**Review by yasaichi — COMMENTED** [review]
See also: https://github.com/necojackarc/breadcrumbs_on_rails-json_ld/pull/1#issuecomment-364424485

**Comment by necojackarc** [comment]
@yasaichi I've updated the PR. please take a look if you have a chance. thanks!

**Review by yasaichi — APPROVED** [review]
LGTM.  Thanks for accepting my suggestion :bow:

**Pull request merged**